File: src/arduinobot_cpp_examples/src/simple_subscriber.cpp
/**
 * @file simple_subscriber.cpp
 * @author vaibhav
 * @brief
 * @version 0.1
 * @date 2023-11-02
 *
 *
 */

#include "arduinobot_cpp_examples/simple_subscriber.h"

SimpleSubscriber::SimpleSubscriber() : Node("simple_subscriber_cpp")
{
    sub_ = this->create_subscription<std_msgs::msg::String>("chatter", 10,
                                                            std::bind(&SimpleSubscriber::subCallback, this, std::placeholders::_1));
}

void SimpleSubscriber::subCallback(const std_msgs::msg::String::SharedPtr msg)
{
    RCLCPP_INFO_STREAM(this->get_logger(), "Received: " << msg->data.c_str());
}

File: src/arduinobot_py_examples/arduinobot_py_examples/simple_subscriber.py
# /usr/bin/env python3

from rclpy.node import Node
from std_msgs.msg import String


class SimpleSubsriber(Node):
    def __init__(self):
        super().__init__("simple_subscriber_py")

        self.sub_ = self.create_subscription(String, "chatter", self.subCallback, 10)

    def subCallback(self, msg: String):
        self.get_logger().info("Received msg: %s" % msg.data)
